inference-engine/src/mkldnn_plugin/nodes/mkldnn_quantize_node.cpp

```diff
@@ -234,9 +235,10 @@ void MKLDNNQuantizeNode::init() {
             float ol = outputLowData[isOutputLowBroadcasted ? 0 : i];
             float oh = outputHighData[isOutputHighBroadcasted ? 0 : i];
 
-            if (ol == oh) {
-                if (levels != 2)
-                    THROW_IE_EXCEPTION << "Quantize layer with name '" << getName() << "' has wrong output quantize ranges";
+            // TODO: ranges shouldn't be equal to INF as well, but for now this check is disabled in order to avoid regressions
+            if (std::isnan(ol) || std::isnan(oh)) {
```

Comment:
why didn't you add such check ```(ol == io && levels != 2)``` for output ranges?

Reply:
From computational standpoint we shouldn't allow this only for input ranges because it uses in following formulas:
`
inputScale[i] = (levels - 1) / (ih - il);
inputShift[i] = -il * (levels - 1) / (ih - il);
`
So il == ih will lead to division on zero.
For output ranges we use another formulas:
`
outputScale[i] = (oh - ol) / (levels - 1);
outputShift[i] = ol;
`
Which means we don't have such limitation.